acquia/mc-cs-plugin-custom-objects
# Injecting objects and fields into forms and columns into reports

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 


:warning: This PR needs changes from PRs [#298](https://github.com/acquia/mc-cs-plugin-custom-objects/pull/298), [#308](https://github.com/acquia/mc-cs-plugin-custom-objects/pull/308), [#316](https://github.com/acquia/mc-cs-plugin-custom-objects/pull/316) so it can be tested after merging them.

#### Description:


This PR injects objects and fields into the form field mapping form and object columns into the form result report.
#### Steps to test this PR:


1. Open this PR for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Setup a custom object. Use Mautic v5.
3. Create a custom object and add new items.
4. Create a form with fields mapped to the custom object. Add fields with text and select types.
    
![image](https://user-images.githubusercontent.com/96085911/216948469-d2d6b8b4-4b2b-40e0-8728-84b441365c28.png)

6. Ensure that mapping works well.
7. Submit the form.
8. Create a new report with the form result as a data source.
![image](https://user-images.githubusercontent.com/96085911/216949296-0b6e8f1a-5794-4cfd-b7a8-6ab5810ae98a.png)
10. Check if all object columns were added.
![image](https://user-images.githubusercontent.com/96085911/216949633-10ea7c9e-842f-43bd-8a70-517866c73f1d.png)
12. Select the fields.
13. Save rthe eport.
14. Check if the repotr is working correctly.
